src/api/types.js

```diff
@@ -17,3 +18,29 @@
  * @import { Request } from '@hapi/hapi'
  * @import { Db } from 'mongodb'
  */
+
+/**
+ * @template T
+ * @typedef {object} Result
+ * @property {T[]} data - The array of data items.
+ * @property {Meta} meta - The metadata about the result.
+ */
+
+/**
+ * @typedef {object} Meta
+ * @property {PaginationResult} [pagination] - The pagination details.
+ */
+
+/**
+ * @typedef {object} PaginationResult
+ * @property {number} page - The current page number.
+ * @property {number} perPage - The number of items per page.
+ * @property {number} totalItems - The total number of items available.
+ * @property {number} totalPages - The total number of pages available.
+ */
```

Comment:
In two minds whether this is the best place for this, or alternatively somewhere in `@defra/forms-model` as will be useful when typing the api result in https://github.com/DEFRA/forms-designer/blob/57fdd49d2e9684e22fca41a24dcd278b024b5aa2/designer/server/src/lib/forms.js
Wdyt @alexluckett @davidjamesstone @colinrotherham ?

Reply:
Yeah good thinking

Whenever we need types in 2+ places stick it in the **forms-designer** model workspace

Reply:
that's opened here: https://github.com/DEFRA/forms-designer/pull/613

Reply:
Agreed